components/plugin/src/miniboxing/plugin/transform/interop/commit/InteropCommitTreeTransformer.scala

```diff
@@ -159,6 +159,15 @@ trait InteropCommitTreeTransformer extends TypingTransformers {
             val res = localTyper.typedOperator(tree2)
              res
 
+          case tree@Apply(apply, List(_ ,evidence)) if apply.symbol == ArrayModule_genericApply =>
+            val targs = tree.tpe.widen.typeArgs
+            if (targs.length == 1) {
+              val targ = targs.head
+              if (flags.flag_warn_mbarrays && targ.typeSymbol.deSkolemize.hasAnnotation(MiniboxedClass))
+                minibox.suboptimalCodeWarning(tree0.pos, minibox.UseMbArrayInsteadOfArrayWarning.msg)
+            }
+            super.transform(tree0)
+
```

Comment:
The problem if I put it there is that Position of symbol will always  be NoPosition and i couldn't show the warning so that is the reason I had to move it to ``InteropCommitTreeTransformer``